docs/ldml/tr35-keyboards.md

```diff
@@ -308,6 +308,263 @@ For extensibility, the `<special>` element will be allowed at nearly every level
+
+#### Example Normalization with Markers
+
+**Example 1**
+
+Consider this example, without markers:
+
+- `e\u{0300}\u{0320}` (input)
+- `e\u{0320}\u{0300}` (NFD)
+
+The combining marks are reordered.
+
+**Example 2**
+
+If we add markers:
+
+- `e\u{0300}\m{marker}\u{0320}` (input)
+- `e\m{marker}\u{0320}\u{0300}` (NFD)
+
+Note that the marker is 'glued' to the _following_ character. In the above example, `\m{marker}` was 'glued' to the `\u{0320}`.
+
+**Example 2**
+
+A second example:
+
+- `e\m{marker0}\u{0300}\m{marker1}\u{0320}\m{marker2}` (input)
+- `e\m{marker1}\u{0320}\m{marker0}\u{0300}\m{marker2}` (NFD)
+
+Here `\m{marker2}` is 'glued' to the end of the string. However, if additional text is added such as by a subsequent keystroke (which may add an additional combining character, for example), this marker may be 'glued' to that following text.
+
+Markers remain in the same normalization-safe segment during normalization. Consider:
+
+**Example 3**
+
+- `e\u{0300}\m{marker1}\u{0320}a\u{0300}\m{marker2}\u{0320}` (original)
+- `e\m{marker1}\u{0320}\u{0300}a\m{marker2}\u{0320}\u{0300}` (NFD)
+
+There are two normalization-safe segments here:
+
+1. `e\u{0300}\m{marker1}\u{0320}`
+2. `a\u{0300}\m{marker2}\u{0320}`
+
+Normalization (and marker rearranging) effectively occurs within each segment.  While `\m{marker1}` is 'glued' to the `\u{0320}`, it is glued within the first segment and has no effect on the second segment.
+
+### Normalization and Character Classes
+
+If pre-composed (non-NFD) characters are used in [character classes](#regex-like-syntax), such as `[á-é]`, these may not match as keyboard authors expect, as the U+00E1 character will not occur in NFD form. Thus this may be masking serious errors in the data.
+
+By default, tools that process the keyboard data should raise an error when character classes include non-NFD characters.
+
+The above should be written instead as a regex `(á|â|ã|ä|å|æ|ç|è|é)`, or as a set variable `<set id="Example" value="á â ã ä å æ ç è é"/>` and matched as `$[Example]`.
+
+There is another case where there is no explicit mention of a non-NFD character, but the character class could include non-NFD characters, such as the range `[\u{0020}-\u{01FF}]`. For these, the tools should raise a warning by default.
+
+### Normalization and Reorders
+
+[`reorder`](#element-reorder) elements operate on NFD codepoints.
+
+### Normalization and Output
+
+On output, text will be normalized into the form requested by that implementation, or possibly specifically requested by a particular application.
+For example, many platforms may request NFC as the output format. In such a case, all text emitted via the keyboard will be transformed into NFC.
+
+Existing text in a document will only have normalization applied within a single normalization-safe segment from the caret.  Output will not contain any markers, thus any normalization is unaffected by any markers embedded within the segment.
```

Comment:
I'm seeing inconsistent line endings on paragraphs. Will you be wrapping at n columns or all on one line? 

(My preference is to wrap even though that introduces more noise in diffs, because it is easier to read in naive text editors)

Reply:
I'm trying not to mutate them for lines that I'm not editing. 

We have a proposal somewhere to put each sentence on its own line, so diffs are clearer.

I could word wrap it late in the release cycle perhaps. Maybe even a release task.